### DISCUSSION TITLE
Navigation problem, stacking instead of changing route

### DISCUSSION THREAD

**Discussion by joshuavachon25** [sveltejs/kit#7366] [Q&A] [answered]
When I click some links on my website, instead of navigating, the content stack stack... I tried to put a #key around my main layout's slot, but even when the key is $page.url.pathname, the bug is still there.

To see the bug, go to the footer and click on Accueil: 
The page will not change but if you go down, you'll see the footer and then the homepage...

Bugs seems to disapear when being at top of the site or if smooth-navigation is desactivated

**Comment by infovore** [comment, marked as answer]
This might well be related to `smooth-navigation` - I'm not fully aware of that library, but I think my answer is related. I've had issues exactly like this when I've had `transition` kicking around in components. What happens, of course, is that when the container block is destroyed, rather than disappearing, the transition plays out, and then the element isn't later swapped out. With my own components using `transition`, the answer has always been to [scope the transition locally](https://svelte.dev/tutorial/local-transitions) - `transition|local`. This means that the navigation doesn't trigger a transition, and the swap-out of the DOM content works correctly.

So: this explains what's going on, but doesn't offer a way to solve it with the desired outcome: working 'smooth' page transitions whilst using the defaul approach to navigation.
